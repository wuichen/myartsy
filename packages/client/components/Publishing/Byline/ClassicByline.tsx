import { color } from "@artsy/palette"
import { avantgarde } from "Assets/Fonts"
import React from "react"
import styled from "styled-components"
import { getAuthorByline, getDate } from "../Constants"
import { ArticleData } from "../Typings"

interface ClassicBylineProps {
  date?: string
  article?: ArticleData
}

export const ClassicByline: React.SFC<ClassicBylineProps> = props => {
  const {
    article: { contributing_authors, author, published_at },
    date,
  } = props

  const contributors = getAuthorByline(contributing_authors, false)

  return (
    <ClassicBylineContainer>
      {contributors ? (
        <div>
          <TextSm>{author.name}</TextSm>
          <div>{`By ${contributors}`}</div>
        </div>
      ) : (
        author.name
      )}
      <TextSm color={color("black60")}>{getDate(date || published_at)}</TextSm>
    </ClassicBylineContainer>
  )
}

const TextSm = styled.div.attrs<{ color?: string }>({})`
  ${avantgarde("s11")}
  color: ${props => (props.color ? props.color : "black")};
`

const ClassicBylineContainer = styled.div`
  display: block;
  ${avantgarde("s13")};
`
