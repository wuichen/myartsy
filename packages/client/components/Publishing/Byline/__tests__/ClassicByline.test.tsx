import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import {
  ClassicArticle,
  ClassicArticleManyAuthors,
} from "../../Fixtures/Articles"
import { ClassicByline } from "../ClassicByline"

describe("ClassicByline", () => {
  describe("Snapshots", () => {
    it("renders a single author", () => {
      const snapshot = renderer.create(
        <ClassicByline article={ClassicArticleManyAuthors} />
      )
      expect(snapshot).toMatchSnapshot()
    })

    it("renders multiple authors", () => {
      const snapshot = renderer.create(
        <ClassicByline article={ClassicArticleManyAuthors} />
      )
      expect(snapshot).toMatchSnapshot()
    })

    it("renders a custom date", () => {
      const snapshot = renderer.create(
        <ClassicByline
          article={ClassicArticleManyAuthors}
          date={"2017-05-19T13:09:18.567Z"}
        />
      )
      expect(snapshot).toMatchSnapshot()
    })
  })

  describe("Unit", () => {
    const getWrapper = _props => {
      return mount(<ClassicByline {..._props} />)
    }
    let props
    beforeEach(() => {
      props = {
        article: ClassicArticle,
      }
    })

    it("renders a single author", () => {
      const component = getWrapper(props)
      expect(component.text()).toMatch("Joanne Artman Gallery")
    })

    it("renders multiple authors", () => {
      props.article = ClassicArticleManyAuthors
      const component = getWrapper(props)
      expect(component.text()).toMatch("Joanne Artman Gallery")
      expect(component.text()).toMatch(
        "By First Author, Second Author and Third Author"
      )
    })

    it("renders published date", () => {
      const component = getWrapper(props)
      expect(component.text()).toMatch("Jul 28, 2017 4:38pm")
    })

    it("renders a custom date", () => {
      props.date = "2017-05-19T13:09:18.567Z"
      const component = getWrapper(props)
      expect(component.text()).toMatch("May 19, 2017 9:09am")
    })
  })
})
