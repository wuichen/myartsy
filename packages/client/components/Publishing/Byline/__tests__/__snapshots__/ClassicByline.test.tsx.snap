// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassicByline Snapshots renders a custom date 1`] = `
.c1 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: black;
}

.c2 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: #666;
}

.c0 {
  display: block;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
}

<div
  className="c0"
>
  <div>
    <div
      className="c1"
    >
      Joanne Artman Gallery
    </div>
    <div>
      By First Author, Second Author and Third Author
    </div>
  </div>
  <div
    className="c2"
    color="#666"
  >
    May 19, 2017 9:09am
  </div>
</div>
`;

exports[`ClassicByline Snapshots renders a single author 1`] = `
.c1 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: black;
}

.c2 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: #666;
}

.c0 {
  display: block;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
}

<div
  className="c0"
>
  <div>
    <div
      className="c1"
    >
      Joanne Artman Gallery
    </div>
    <div>
      By First Author, Second Author and Third Author
    </div>
  </div>
  <div
    className="c2"
    color="#666"
  >
    Jul 28, 2017 4:38pm
  </div>
</div>
`;

exports[`ClassicByline Snapshots renders multiple authors 1`] = `
.c1 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: black;
}

.c2 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: #666;
}

.c0 {
  display: block;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
}

<div
  className="c0"
>
  <div>
    <div
      className="c1"
    >
      Joanne Artman Gallery
    </div>
    <div>
      By First Author, Second Author and Third Author
    </div>
  </div>
  <div
    className="c2"
    color="#666"
  >
    Jul 28, 2017 4:38pm
  </div>
</div>
`;
